Openstack inventory updates from Tower.

cchurch opened the pull request:
Includes changes from https://github.com/ansible/ansible/pull/10491 and https://github.com/ansible/ansible/pull/10776 and allows specifying a config file via environment variable.

abadger commented on plugins/inventory/openstack.py:
```diff
@@ -152,8 +156,7 @@ def main():
     except shade.OpenStackCloudException as e:
-        print(e.message)
-        sys.exit(1)
+        sys.exit(e.message)
```
I'd rather be explicit and use:

``` python
sys.stderr.write('%s\n' % e.message)
sys.exit(1)
```

cchurch replied:
Pushed a new commit to fix.  Looking at the other changes in https://github.com/ansible/ansible/pull/10776, that PR is really inconsistent in this regard.

abadger commented:
@emonty Care to review this and +1 if it looks good to merge?

emonty commented on plugins/inventory/openstack.py:
```diff
@@ -51,11 +51,14 @@
     def __init__(self, private=False, refresh=False):
+        config_files = os_client_config.config.CONFIG_FILES
+        if os.environ.get('OPENSTACK_CONFIG_FILE', None):
+            config_files.insert(0, os.environ['OPENSTACK_CONFIG_FILE'])
```
Would you be amenable to changing this to "OS_CLIENT_CONFIG_FILE"? If so, I'd like to upstream support for setting this into os-client-config itself (it seems like a generally useful feature) We can land it here safely (with the name change) until we get it landed there.

emonty commented:
I created a commit for supporting the env-var oerride. in openstack/os-client-config here: https://review.openstack.org/198502.

emonty commented:
@cchurch you ok with the env var name change? We've landed the support for it generally in os-client-config and I can cut a release asap if you're good with that approach.

cchurch committed:
Openstack inventory updates from Tower.  Includes changes from https://github.com/ansible/ansible/pull/10491 and https://github.com/ansible/ansible/pull/10776.

cchurch commented:
@emonty I've removed my environment variable override since it's now supported by os-client-config.

@abadger @bcoca I've rebased & squashed, should be ready to go.
